nav2_tasks/include/nav2_tasks/task_server.hpp

```diff
@@ -94,17 +94,22 @@ class TaskServer : public rclcpp::Node
       if (cvCommand_.wait_for(lock, std::chrono::milliseconds(100),
         [&] {return commandReceived_ == true;}))
       {
+        nav2_msgs::msg::TaskStatus statusMsg;
+	    TaskStatus status = TaskStatus::FAILED;
+
         // Call the user's overridden method
-        TaskStatus status = execute(commandMsg_);
+        try {
+          status = execute(commandMsg_);
+        } catch (...) {
+          statusMsg.result = nav2_msgs::msg::TaskStatus::FAILED;
```

Comment:
This is important as well, but it's not exactly what I was looking for. I was looking for something more like this:
https://stackoverflow.com/questions/233127/how-can-i-propagate-exceptions-between-threads

With the change as written, the exception details are lost, and the worker thread keeps running.

I think we want something more like:
```c++
} catch (...) {
  exception_ptr_ = std::current_exception();
  statusMsg.result = nav2_msgs::msg::TaskStatus::FAILED;
  statusPub_->publish(statusMsg);
  return;
}
```
and then also:
```c++
void stopWorkerThread()
  {
    workerThread_->join();
    if (exception_ptr_ != nullptr)
      std::rethrow_exception(exception_ptr_);
    delete workerThread_;
    workerThread_ = nullptr;
}
```

Reply:
I think of it like this: There are three "actors" here: the client, the task server (handling communication with the client and interfacing to the user's execute code), and the user's execute code. The client has invoked an operation which will return SUCCESS, FAILURE, or CANCELED. The task server's job is to invoke the execute code and return the result. If the user's execute code should generate an exception for a particular invocation of a task, I don't think that the task server should exit its operation (quit the thread). It can return failure and be ready to be invoked again. 

From a developer's point of view, we definitely need to understand what happened. The log messages provide that info. From the runtime perspective, I don't see that propagating the exception up higher would allow any kind of recovery or improved behavior. Instead, the task server would effectively be dead and couldn't be called again. It would have to be reloaded.

Reply:
The way I see it, if the user's execute code should handle any exceptions they want to handle. If an exception makes it all the way to the task server code, we need to halt and report the problem. 

The task server should not be deciding the exception handling policy for the user's execute code.

As a user of the task server library, that is the behavior I want.

Reply:
Let's have a discussion to hash out the issues and summarize the results here. I'd like to talk about the interface to provide to task clients, how this intersects recovery, etc.

Reply:
Result of the discussion was that we'll implement it per Carl's suggestion. That is, propagate the exception to the thread that owns the TaskServer object. The TaskServer template will wrap the call to execute in a try/catch block. Upon an exception, it will save the exception in a member variable. The next job is to get the object to stop spinning so that it can reach its destructor and propagate the exception. For now, we'll use rclcpp::shutdown, but it would be better if there was a way to get this node to stop spinning. In the TaskServer destructure the exception will be propagated.